### ISSUE TITLE
[EKS] [request]: Support/Document using mirantis/cri-dockerd in EKS 1.23+

### ISSUE THREAD

**Issue by WilliamABradley** [aws/containers-roadmap#1514]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
What do you want us to build?

At least add documentation on how to install and run EKS with cri-dockerd.
An extra beneficial step would be to have to option to opt for cri-dockerd out of the box.

**Which service(s) is this request for?**
This could be Fargate, ECS, EKS, ECR

EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
What outcome are you trying to achieve, ultimately, and why is it hard/impossible to do right now? What is the impact of not having this problem solved? The more details you can provide, the better we'll be able to understand and solve the problem.

We use EKS as a devops cluster, and having access to the docker socket is hugely beneficial for being able to build docker images, as well as run tests via Docker compose. It would be very helpful to be able to continue to use these tools (via cri-dockerd) without having to rewrite all of our devops to use alternatives like img, and kompose.

**Are you currently working around this issue?**
How are you currently solving this problem?

We are currently using Kubernetes 1.21, and will opt for the container runtime flag in our managed node groups, to maintain Docker CRI until 1.22.

**Additional context**
Anything else we should know?

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by stevehipwell** [comment]
Have you looked at moving to [Kaniko](https://github.com/GoogleContainerTools/kaniko) for your container builds? Alternatively you could build your own AMI to do this, or even customize the default AMI with userdata in the launch template (I'm not advising you do this).

**Comment by stevehipwell** [comment]
@WilliamABradley I don't think this pattern should be officially documented, it's only purpose is to enable/encourage the anti-pattern of running a privileged container and exposing the Docker socket. We should be moving people away from this pattern and not encouraging it.

RE CI/CD tools, there are plenty which work correctly in K8s without needing a privileged security context and a Docker socket.

**Comment by dims** [comment]
Agree with @stevehipwell here. Closing it.
